src/main/java/com/microsoft/sqlserver/jdbc/SQLServerClob.java

```diff
@@ -281,9 +280,20 @@ public InputStream getAsciiStream() throws SQLException {
         if (null != sqlCollation && !sqlCollation.supportsAsciiConversion())
             DataTypes.throwConversionError(getDisplayClassName(), "AsciiStream");
 
-        getStringFromStream();
-        InputStream getterStream = new BufferedInputStream(
-                new ReaderInputStream(new StringReader(value), US_ASCII, value.length()));
+        // Need to use a BufferedInputStream since the stream returned by this method is assumed to support mark/reset
+        InputStream getterStream;
+        if (null == value && !activeStreams.isEmpty()) {
+            InputStream inputStream = (InputStream) activeStreams.get(0);
+            try {
+                inputStream.reset();
+                getterStream = inputStream;
```

Comment:
Can we close 'inputStream' here?

Reply:
It's the same input stream that the user gets, if we close it, the stream returned to the user gets closed.